crates/runtime/src/instance/allocator/pooling/index_allocator.rs

```diff
@@ -80,7 +86,7 @@ struct Inner {
     ///
     /// The `List` here is appended to during deallocation and removal happens
     /// from the tail during allocation.
-    module_affine: HashMap<CompiledModuleId, List>,
+    module_affine: HashMap<(CompiledModuleId, DefinedMemoryIndex), List>,
```

Comment:
note from chris: can we newtype this tuple? `ModuleAndMemory` or something like that